CD: Sync seed/migrate execution order

SarahJaine opened the pull request:
**Description of change**
In https://github.com/adhocteam/Head-Start-TTADP/pull/94 I was banking on tasks being put in a synchronous queue for execution. This proved to not be how cf handles task execution. As evidenced on the merge to the feature branch, the seed task failed because migration was not run first.  This PR puts the migration and seed commands in a single task to ensure correct execution order.

**How to test**
after merging this PR to the feature branch check the sandbox log stream, you should see all the db tasks run successfully (undo seed, undo migrate, migrate, seed). I verified the database transformations worked on my last deploy to sandbox.


**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/52

**Checklist**
<!-- Add details to each completed item -->
- [x] Meets issue criteria
- [x] Code tested
- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [x] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

jasalisbury commented on .circleci/config.yml:
```diff
@@ -117,8 +117,8 @@ commands:
-            cf run-task << parameters.app_name >> "yarn db:migrate:prod"
-            cf run-task << parameters.app_name >> "yarn db:seed:prod"
+            cf run-task << parameters.app_name >> \
+            "yarn db:migrate:prod && yarn db:seed:prod"
```
I think we might still run into a race... We might have to do `yarn db:seed:undo && yarn db:migrate:undo &&  yarn db:migrate && yarn db:seed` in one task. Otherwise we could start migrating the db before the `undo` task has completed (or started).

SarahJaine replied:
I was thinking this wouldn't create a race condition because of circle ci job step order and db locking. In the steps, the undo seed/migrate task is submitted to cf and then cf has to respond that it successfully submitted the task. After circle ci gets this success message back, it starts the next step of the job, submitting the seed/migrate task. By then the db should be locked right? I know sequelize (and node in general) are built on async, but running migrations should enforce some order right? You can't run migrate and undo migrate at the same time, right?

jasalisbury replied:
I'm not sure what locking goes on, but regardless it seems there is at least chance between `seed:undo` and `migrate:undo` for the `db:migrate` (from the second step) to fire off. Not only that but these jobs are just submitted to cf so it seems possible for them to run out of order (even though the "undo" job hits the queue first) depending on how the first job is "queued".

SarahJaine replied:
Yeah you're probably right. (It was just more elegant the other way- chaining commands seems so hacky.) But yeah, that'll ensure proper execution order and prevent devs from having to go check cf application logs- which is what's actually important.

jasalisbury replied:
> (It was just more elegant the other way- chaining commands seems so hacky.)

Yeah I definitely agree!

SarahJaine commented:
Last cf application log output on sandbox demonstrates successful ordering of yarn db commands
```
11:07:49.924: [APP/TASK/f1a60a22.0] yarn run v1.22.10
11:07:49.969: [APP/TASK/f1a60a22.0] $ node_modules/.bin/sequelize db:seed:undo:all --options-path .production.sequelizerc
11:07:50.263: [APP/TASK/f1a60a22.0] Sequelize CLI [Node: 12.20.0, CLI: 6.2.0, ORM: 5.22.3]
11:07:50.269: [APP/TASK/f1a60a22.0] Loaded configuration file "build/config/config.js".
11:07:50.269: [APP/TASK/f1a60a22.0] Using environment "production".
11:07:50.368: [APP/TASK/f1a60a22.0] == 20201124160449-users: reverting =======
11:07:50.401: [APP/TASK/f1a60a22.0] == 20201124160449-users: reverted (0.033s)
11:07:50.405: [APP/TASK/f1a60a22.0] Done in 0.48s.
11:07:50.593: [APP/TASK/f1a60a22.0] $ node_modules/.bin/sequelize db:migrate:undo:all --options-path .production.sequelizerc
11:07:50.553: [APP/TASK/f1a60a22.0] yarn run v1.22.10
11:07:50.866: [APP/TASK/f1a60a22.0] Sequelize CLI [Node: 12.20.0, CLI: 6.2.0, ORM: 5.22.3]
11:07:50.872: [APP/TASK/f1a60a22.0] Loaded configuration file "build/config/config.js".
11:07:50.872: [APP/TASK/f1a60a22.0] Using environment "production".
11:07:51.038: [APP/TASK/f1a60a22.0] == 20201206184519-rename-user-title-column: reverting =======
11:07:51.078: [APP/TASK/f1a60a22.0] == 20201206184519-rename-user-title-column: reverted (0.040s)
11:07:51.085: [APP/TASK/f1a60a22.0] == 20201206173716-add-default-createdat-updatedat: reverting =======
11:07:51.104: [APP/TASK/f1a60a22.0] == 20201206173716-add-default-createdat-updatedat: reverted (0.019s)
11:07:51.111: [APP/TASK/f1a60a22.0] == 20201205200638-create-grant-goal: reverting =======
11:07:51.207: [APP/TASK/f1a60a22.0] == 20201123143446-add-unique-constraint-to-roletopic: reverting =======
11:07:51.126: [APP/TASK/f1a60a22.0] == 20201205200638-create-grant-goal: reverted (0.015s)
11:07:51.131: [APP/TASK/f1a60a22.0] == 20201205200637-create-grant: reverting =======
11:07:51.141: [APP/TASK/f1a60a22.0] == 20201205200637-create-grant: reverted (0.009s)
11:07:51.147: [APP/TASK/f1a60a22.0] == 20201130144748-create-grantee: reverting =======
11:07:51.157: [APP/TASK/f1a60a22.0] == 20201130144748-create-grantee: reverted (0.010s)
11:07:51.162: [APP/TASK/f1a60a22.0] == 20201130144721-add-unique-constraint-to-topicgoal: reverting =======
11:07:51.171: [APP/TASK/f1a60a22.0] == 20201130144721-add-unique-constraint-to-topicgoal: reverted (0.008s)
11:07:51.176: [APP/TASK/f1a60a22.0] == 20201125195340-create-topic-goal: reverting =======
11:07:51.186: [APP/TASK/f1a60a22.0] == 20201125195340-create-topic-goal: reverted (0.010s)
11:07:51.192: [APP/TASK/f1a60a22.0] == 20201125194528-create-goal: reverting =======
11:07:51.201: [APP/TASK/f1a60a22.0] == 20201125194528-create-goal: reverted (0.009s)
11:07:51.215: [APP/TASK/f1a60a22.0] == 20201123143446-add-unique-constraint-to-roletopic: reverted (0.009s)
11:07:51.221: [APP/TASK/f1a60a22.0] == 20201120183631-create-role-topic: reverting =======
11:07:51.230: [APP/TASK/f1a60a22.0] == 20201120183631-create-role-topic: reverted (0.010s)
11:07:51.236: [APP/TASK/f1a60a22.0] == 20201120182130-create-role: reverting =======
11:07:51.246: [APP/TASK/f1a60a22.0] == 20201120182130-create-role: reverted (0.009s)
11:07:51.252: [APP/TASK/f1a60a22.0] == 20201120133733-create-topic: reverting =======
11:07:51.262: [APP/TASK/f1a60a22.0] == 20201120133733-create-topic: reverted (0.010s)
11:07:51.269: [APP/TASK/f1a60a22.0] == 20201102125023-create-request-errors: reverting =======
11:07:51.279: [APP/TASK/f1a60a22.0] == 20201102125023-create-request-errors: reverted (0.010s)
11:07:51.284: [APP/TASK/f1a60a22.0] == 20201030204458-add-unique-constraint-to-permissions: reverting =======
11:07:51.292: [APP/TASK/f1a60a22.0] == 20201030204458-add-unique-constraint-to-permissions: reverted (0.008s)
11:07:51.297: [APP/TASK/f1a60a22.0] == 20201029214432-add-title-to-user: reverting =======
11:07:51.314: [APP/TASK/f1a60a22.0] == 20201029214432-add-title-to-user: reverted (0.017s)
11:07:51.408: [APP/TASK/f1a60a22.0] == 20200827192413-create-user: reverting =======
11:07:51.319: [APP/TASK/f1a60a22.0] == 20201029214146-create-permission: reverting =======
11:07:51.331: [APP/TASK/f1a60a22.0] == 20201029214146-create-permission: reverted (0.012s)
11:07:51.336: [APP/TASK/f1a60a22.0] == 20201029143000-scope: reverting =======
11:07:51.359: [APP/TASK/f1a60a22.0] == 20201029143000-scope: reverted (0.022s)
11:07:51.365: [APP/TASK/f1a60a22.0] == 20201029143000-region: reverting =======
11:07:51.374: [APP/TASK/f1a60a22.0] == 20201029143000-region: reverted (0.009s)
11:07:51.378: [APP/TASK/f1a60a22.0] == 20201029142912-create-region: reverting =======
11:07:51.388: [APP/TASK/f1a60a22.0] == 20201029142912-create-region: reverted (0.009s)
11:07:51.394: [APP/TASK/f1a60a22.0] == 20201029141201-create-scope: reverting =======
11:07:51.404: [APP/TASK/f1a60a22.0] == 20201029141201-create-scope: reverted (0.009s)
11:07:51.417: [APP/TASK/f1a60a22.0] == 20200827192413-create-user: reverted (0.009s)
11:07:51.422: [APP/TASK/f1a60a22.0] Done in 0.87s.
11:07:51.567: [APP/TASK/f1a60a22.0] yarn run v1.22.10
11:07:51.610: [APP/TASK/f1a60a22.0] $ node_modules/.bin/sequelize db:migrate --options-path .production.sequelizerc
11:07:51.885: [APP/TASK/f1a60a22.0] Sequelize CLI [Node: 12.20.0, CLI: 6.2.0, ORM: 5.22.3]
11:07:51.891: [APP/TASK/f1a60a22.0] Loaded configuration file "build/config/config.js".
11:07:52.020: [APP/TASK/f1a60a22.0] == 20200827192413-create-user: migrating =======
11:07:51.891: [APP/TASK/f1a60a22.0] Using environment "production".
11:07:52.046: [APP/TASK/f1a60a22.0] == 20200827192413-create-user: migrated (0.026s)
11:07:52.053: [APP/TASK/f1a60a22.0] == 20201029141201-create-scope: migrating =======
11:07:52.069: [APP/TASK/f1a60a22.0] == 20201029141201-create-scope: migrated (0.016s)
11:07:52.076: [APP/TASK/f1a60a22.0] == 20201029142912-create-region: migrating =======
11:07:52.088: [APP/TASK/f1a60a22.0] == 20201029142912-create-region: migrated (0.012s)
11:07:52.094: [APP/TASK/f1a60a22.0] == 20201029143000-region: migrating =======
11:07:52.104: [APP/TASK/f1a60a22.0] == 20201029143000-region: migrated (0.009s)
11:07:52.110: [APP/TASK/f1a60a22.0] == 20201029143000-scope: migrating =======
11:07:52.212: [APP/TASK/f1a60a22.0] == 20201120133733-create-topic: migrating =======
11:07:52.120: [APP/TASK/f1a60a22.0] == 20201029143000-scope: migrated (0.009s)
11:07:52.125: [APP/TASK/f1a60a22.0] == 20201029214146-create-permission: migrating =======
11:07:52.143: [APP/TASK/f1a60a22.0] == 20201029214146-create-permission: migrated (0.017s)
11:07:52.149: [APP/TASK/f1a60a22.0] == 20201029214432-add-title-to-user: migrating =======
11:07:52.167: [APP/TASK/f1a60a22.0] == 20201029214432-add-title-to-user: migrated (0.018s)
11:07:52.175: [APP/TASK/f1a60a22.0] == 20201030204458-add-unique-constraint-to-permissions: migrating =======
11:07:52.186: [APP/TASK/f1a60a22.0] == 20201030204458-add-unique-constraint-to-permissions: migrated (0.011s)
11:07:52.191: [APP/TASK/f1a60a22.0] == 20201102125023-create-request-errors: migrating =======
11:07:52.206: [APP/TASK/f1a60a22.0] == 20201102125023-create-request-errors: migrated (0.014s)
11:07:52.226: [APP/TASK/f1a60a22.0] == 20201120133733-create-topic: migrated (0.015s)
11:07:52.234: [APP/TASK/f1a60a22.0] == 20201120182130-create-role: migrating =======
11:07:52.247: [APP/TASK/f1a60a22.0] == 20201120182130-create-role: migrated (0.014s)
11:07:52.254: [APP/TASK/f1a60a22.0] == 20201120183631-create-role-topic: migrating =======
11:07:52.266: [APP/TASK/f1a60a22.0] == 20201120183631-create-role-topic: migrated (0.013s)
11:07:52.272: [APP/TASK/f1a60a22.0] == 20201123143446-add-unique-constraint-to-roletopic: migrating =======
11:07:52.283: [APP/TASK/f1a60a22.0] == 20201123143446-add-unique-constraint-to-roletopic: migrated (0.010s)
11:07:52.289: [APP/TASK/f1a60a22.0] == 20201125194528-create-goal: migrating =======
11:07:52.307: [APP/TASK/f1a60a22.0] == 20201125194528-create-goal: migrated (0.018s)
11:07:52.313: [APP/TASK/f1a60a22.0] == 20201125195340-create-topic-goal: migrating =======
11:07:52.326: [APP/TASK/f1a60a22.0] == 20201125195340-create-topic-goal: migrated (0.013s)
11:07:52.333: [APP/TASK/f1a60a22.0] == 20201130144721-add-unique-constraint-to-topicgoal: migrating =======
11:07:52.344: [APP/TASK/f1a60a22.0] == 20201130144721-add-unique-constraint-to-topicgoal: migrated (0.011s)
11:07:52.350: [APP/TASK/f1a60a22.0] == 20201130144748-create-grantee: migrating =======
11:07:52.363: [APP/TASK/f1a60a22.0] == 20201130144748-create-grantee: migrated (0.013s)
11:07:52.369: [APP/TASK/f1a60a22.0] == 20201205200637-create-grant: migrating =======
11:07:52.388: [APP/TASK/f1a60a22.0] == 20201205200637-create-grant: migrated (0.020s)
11:07:52.396: [APP/TASK/f1a60a22.0] == 20201205200638-create-grant-goal: migrating =======
11:07:52.409: [APP/TASK/f1a60a22.0] == 20201205200638-create-grant-goal: migrated (0.013s)
11:07:52.416: [APP/TASK/f1a60a22.0] == 20201206173716-add-default-createdat-updatedat: migrating =======
11:07:52.612: [APP/TASK/f1a60a22.0] yarn run v1.22.10
11:07:52.433: [APP/TASK/f1a60a22.0] == 20201206173716-add-default-createdat-updatedat: migrated (0.017s)
11:07:52.439: [APP/TASK/f1a60a22.0] == 20201206184519-rename-user-title-column: migrating =======
11:07:52.453: [APP/TASK/f1a60a22.0] == 20201206184519-rename-user-title-column: migrated (0.013s)
11:07:52.458: [APP/TASK/f1a60a22.0] Done in 0.89s.
11:07:52.657: [APP/TASK/f1a60a22.0] $ node_modules/.bin/sequelize db:seed:all --options-path .production.sequelizerc
11:07:52.967: [APP/TASK/f1a60a22.0] Sequelize CLI [Node: 12.20.0, CLI: 6.2.0, ORM: 5.22.3]
11:07:52.973: [APP/TASK/f1a60a22.0] Loaded configuration file "build/config/config.js".
11:07:53.050: [APP/TASK/f1a60a22.0] == 20201124160449-users: migrating =======
11:07:52.973: [APP/TASK/f1a60a22.0] Using environment "production".
11:07:53.137: [CELL.0] Cell 67a9e588-5d4e-4889-9388-b0b743f3ac5a stopping instance dc3cea90-d58d-41ba-a7c5-33771bdf9dd0
11:07:53.064: [APP/TASK/f1a60a22.0] == 20201124160449-users: migrated (0.013s)
11:07:53.068: [APP/TASK/f1a60a22.0] Done in 0.46s.
11:07:53.075: [APP/TASK/f1a60a22.0] Exit status 0
```

jasalisbury reviewed (approved):
:shipit:
